budget_allocation_dimension/models/base_budget_move.py

```diff
@@ -16,11 +16,40 @@ def _get_dimension_fields(self):
             x for x in self.fields_get().keys() if x.startswith("x_dimension_")
         ]
 
-    def _where_query_source_fund(self, docline):
-        where_query = super()._where_query_source_fund(docline)
+    def _get_condition_skip_dimension(self):
+        """
+        This function will check state 2 module
+            1. budget_source_fund
+            2. budget_allocation_dimension_fund
+
+        ============ budget.monitor.report ============
+        - Not install module `budget_source_fund` -> query dimension
+
+        ============ source.fund.monitor.report ============
+        - Install module `budget_source_fund`,
+          Not install module `budget_allocation_dimension_fund` -> skip dimension
+        - Install module `budget_source_fund`,
+          install module `budget_allocation_dimension_fund` -> query dimension
+        """
+        dimension_fund = self.env["ir.module.module"].search(
+            [("name", "=", "budget_allocation_dimension_fund")]
+        )
+        # Skip query dimension
+        if (
+            self.env.get("source.fund.monitor.report", "/") != "/"
+            and dimension_fund.state != "installed"
+        ):
+            return True
+        return False
+
+    def _get_where_commitment(self, docline):
+        where_query = super()._get_where_commitment(docline)
+        skip_query_dimension = self._get_condition_skip_dimension()
+        if skip_query_dimension:
+            return where_query
         dimensions = docline._get_dimension_fields()
         where_dimensions = [
-            "{} {} {}".format(
+            "({0} {1} {2} or {0} = 999999990)".format(
```

Comment:
What is 9999999 for? Where it come from?

Reply:
Fake id on budget control monitor (it don't have x_dimension_ on budget monitoring).


https://github.com/ecosoft-odoo/budgeting/pull/269/files#diff-e729f5389f8397af4acbb8fad15d0bdb763a348dd1fd46812d2e86836389d89d